envoyauth/response.go

```diff
@@ -185,30 +185,67 @@ func (result *EvalResult) GetRequestHTTPHeadersToRemove() ([]string, error) {
 	return result.getStringSliceFromDecision("request_headers_to_remove")
 }
 
-func (result *EvalResult) getHeadersFromDecision(fieldName string) ([]*ext_core_v3.HeaderValueOption, error) {
+func (result *EvalResult) makeHeadersFromDecision(fieldName string, h headersBaggage) error {
 	switch decision := result.Decision.(type) {
 	case bool:
-		return nil, nil
+		return nil
 	case map[string]interface{}:
 		val, ok := decision[fieldName]
 		if !ok {
-			return nil, nil
+			return nil
 		}
 
-		return transformHeadersToEnvoy(val)
+		return transformHeadersToEnvoy(val, h)
 	default:
-		return nil, result.invalidDecisionErr()
+		return result.invalidDecisionErr()
 	}
 }
 
+type headersBaggage interface {
+	allocate(n int)
+	add(k, v string)
+}
+
+type envoyHeaders []*ext_core_v3.HeaderValueOption
+
+func (h *envoyHeaders) add(k, v string) {
+	*h = append(*h, makeHeaderValueOption(k, v))
+}
+
+func (h *envoyHeaders) allocate(n int) {
+	*h = slices.Grow(*h, n)
+}
+
+type httpHeaders struct {
+	http.Header
+}
```

Comment:
Do you need the `struct {}`? Wouldn't one of these do the trick, too? Frankly, I cannot remember which one. 😅 
```go
type httpHeaders = http.Header

type httpHeaders http.Header
```

Reply:
Yes, `http.Header` should be wrapped into a struct.

The following variants don't allow adding new functions.